pkg/stub/handler.go

```diff
@@ -606,23 +607,36 @@ func (h *Handler) CreateDefaultResourceIfNeeded(srcfg *v1alpha1.SamplesResource)
 	return srcfg, nil
 }
 
-func (h *Handler) WaitingForCredential(srcfg *v1alpha1.SamplesResource) error {
+// WaitingForCredential determines whether we should proceed with processing the sample resource event,
+// where we should *NOT* proceed if we are RHEL and using the default redhat registry;  The return from
+// this method is in 2 flavors:  1) if the boolean is true, tell the caller to just return nil to the sdk;
+// 2) after the boolean is checked, if need be, return the error (and subesequent requeue, etc.) associated
+// with setting a message in the import credential condition noting in the import condition we need a credential
+func (h *Handler) WaitingForCredential(srcfg *v1alpha1.SamplesResource) (bool, error) {
 	if srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) {
-		return nil
+		return false, nil
 	}
 
 	// if trying to do rhel to the default registry.redhat.io registry requires the secret
 	// be in place since registry.redhat.io requires auth to pull; since it is not ready
 	// log error state
 	if srcfg.Spec.InstallType == v1alpha1.RHELSamplesDistribution &&
 		(srcfg.Spec.SamplesRegistry == "" || srcfg.Spec.SamplesRegistry == "registry.redhat.io") {
-		// currently do not attempt to avoid multiple settings of this conditions/message prior to an event
-		// where the situation is addressed
+		cred := srcfg.Condition(v1alpha1.ImportCredentialsExist)
+		// - if import cred is false, and the message is empty, that means we have NOT registered the error, and need to do so
+		// - if cred is false, and the message is there, we can just return nil to the sdk, which "true" for the boolean return value indicates;
+		// not returning the same error multiple times to the sdk avoids additional churn; once the secret comes in, it will update the samplesresource
+		// with cred == true, and then we'll get another samplesresource event that will trigger config processing
+		if len(cred.Message) > 0 {
+			return true, nil
+		}
 		err := fmt.Errorf("Cannot create rhel imagestreams to registry.redhat.io without the credentials being available")
 		h.processError(srcfg, v1alpha1.ImportCredentialsExist, corev1.ConditionFalse, err, "%v")
-		return err
+		return false, err
```

Comment:
make this `return true, err`

Reply:
I assume this goes in conjunction with your comment below ...

Reply:
ah yeah ... your next comment starts with `then make ...`